src/schema/schema_open.c

```diff
@@ -746,6 +746,7 @@ __wt_schema_open_oligarch(WT_SESSION_IMPL *session)
     /* Add the ingest table file identifier into the oligarch managers list of tracked tables */
     ingest_id = ((WT_BTREE *)oligarch->ingest->handle)->id;
     stable_id = ((WT_BTREE *)oligarch->stable->handle)->id;
+    WT_ASSERT(session, FLD_ISSET(stable_id, WT_BTREE_ID_NAMESPACE_SHARED));
```

Comment:
Please replace this by masking out the namespace and then doing the comparison, e.g.,
```
WT_ASSERT(session, stable_id & (WT_BTREE_ID_NAMESPACE_BITS - 1) == WT_BTREE_ID_NAMESPACE_SHARED);
```
...except that you probably want to make `stable_id & (WT_BTREE_ID_NAMESPACE_BITS - 1)` to be a macro.

The reason is that if we ever have namespace ID == 3 in the future, it would still have the lowest bit set, and it would thus satisfy the assertion as it is currently written.

Reply:
Thanks! Done in 3b45284.